### PR TITLE
Added parallel axis methods for Particle and RigidBody

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References to other Issues or PRs

Closes #12159 

#### Release Notes

* physics.mechanics
  * `parallel_axis` method has been added to `Particle` and `RigidBody`.

### CODE REVIEW

**Comment by czgdp1807** [comment]
ping @moorepants

**Comment by czgdp1807** [comment]
@sympy/mechanics @moorepants Please take a look at it and let me know if any other changes are required.

**Comment by czgdp1807** [comment]
The tests are failing due to the tensorflow update. We will restart the failed jobs once the master is updated with the fix for the above update.

**Comment by czgdp1807** [comment]
Locally on my system both the tests are passing, but the same is failing on travis. I am unable to figure out this issue. Can someone please help me with this? @sympy/mechanics 
```python
============================================ test process starts =============================================
executable:         /usr/bin/python2  (2.7.15-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       gmpy 1.17
numpy:              1.16.4
random seed:        59147076
hash randomization: on (PYTHONHASHSEED=4004077473)

sympy/physics/mechanics/tests/test_rigidbody.py[5] .....                                                  [OK]

================================= tests finished: 5 passed, in 1.28 seconds ==================================

============================================ test process starts =============================================
executable:         /usr/bin/python3  (3.6.8-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              1.14.5
random seed:        61091225
hash randomization: on (PYTHONHASHSEED=1750012188)

sympy/physics/mechanics/tests/test_rigidbody.py[5] .....                                                  [OK]

================================= tests finished: 5 passed, in 1.04 seconds ==================================                    
```

**Comment by sylee957** [comment]
Have you tried running the test with symengine?
I think that the test is from the symengine tests, as far as I know.
I was also not able to reproduce the issue, but getting some other weird errors, but at least this can be a point to begin searching.

**Comment by czgdp1807** [comment]
> Have you tried running the test with symengine?

How to do that?

**Comment by sylee957** [comment]
You can use the command below as done in the travis
```bash
conda install python-symengine=0.3 -c symengine/label/dev -c conda-forge
export USE_SYMENGINE=1
python bin/test mechanics
```
But I'm not totally sure because symengine is not working for my case.

**Comment by czgdp1807** [comment]
I will merge it if the tests pass as everything looks good.

**Comment by oscarbenjamin** [comment]
It doesn't look like anyone else has reviewed this

**Comment by czgdp1807** [comment]
Yes, I will not merge for at least 24 hours. 
ping @sympy/mechanics

File: sympy/physics/mechanics/particle.py [hunk 1]
```diff
@@ -224,3 +224,35 @@ def potential_energy(self, scalar):
...
 
         self._pe = sympify(scalar)
+
+    def set_potential_energy(self, scalar):
+        SymPyDeprecationWarning(
+                feature="Method sympy.physics.mechanics." +
+                    "Particle.set_potential_energy(self, scalar)",
+                useinstead="property sympy.physics.mechanics." +
+                    "Particle.potential_energy",
+                deprecated_since_version="1.0", issue=9800).warn()
```
Comment (moorepants) [inline sympy/physics/mechanics/particle.py]: This should be the current SymPy version 1.4.

Comment (moorepants) [reply sympy/physics/mechanics/particle.py]: I mean is should be the next one: 1.5.

File: sympy/physics/mechanics/rigidbody.py [hunk 2]
```diff
@@ -297,3 +298,35 @@ def potential_energy(self, scalar):
...
 
         self._pe = sympify(scalar)
+
+    def set_potential_energy(self, scalar):
+        SymPyDeprecationWarning(
+                feature="Method sympy.physics.mechanics." +
+                    "RigidBody.set_potential_energy(self, scalar)",
+                useinstead="property sympy.physics.mechanics." +
+                    "RigidBody.potential_energy",
+                deprecated_since_version="1.0", issue=9800).warn()
```
Comment (moorepants) [inline sympy/physics/mechanics/rigidbody.py]: 1.5

**Comment by czgdp1807** [comment]
@moorepants I have made the updates. Please let me know if something more is to be done. If you support merging then please indicate.

**Comment by moorepants** [comment]
Thanks for working on this!

**Pull request merged**